modules/signals/src/with-computed.ts

```diff
@@ -1,22 +1,49 @@
+import { computed, isSignal, Signal } from '@angular/core';
 import {
-  SignalsDictionary,
   SignalStoreFeature,
   SignalStoreFeatureResult,
   StateSignals,
 } from './signal-store-models';
 import { Prettify } from './ts-helpers';
 import { withProps } from './with-props';
 
+type ComputedResult<
+  ComputedSignals extends Record<
+    string | symbol,
+    Signal<unknown> | (() => unknown)
+  >
+> = {
+  [P in keyof ComputedSignals]: ComputedSignals[P] extends () => infer V
+    ? Signal<V>
+    : ComputedSignals[P];
+};
+
 export function withComputed<
   Input extends SignalStoreFeatureResult,
-  ComputedSignals extends SignalsDictionary
+  ComputedSignals extends Record<
+    string | symbol,
+    Signal<unknown> | (() => unknown)
+  >
 >(
   signalsFactory: (
```

Comment:
we can also rename this:

```suggestion
  computedFactory: (
```

Reply:
Done